components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/impl/ApplicationsApiServiceImpl.java

```diff
@@ -15,29 +25,132 @@
+        //todo: validation, need to be moved
+        String username = CarbonContext.getThreadLocalCarbonContext().getUsername();
+        if (StringUtils.isEmpty(body.getName())) {
+            return Response.serverError().entity(new ApiResponseMessage(ApiResponseMessage.ERROR, "Name is empty"))
+                    .build();
+        }
+        if (StringUtils.isEmpty(body.getThrottlingTier())) {
+            return Response.serverError().entity(new ApiResponseMessage(ApiResponseMessage.ERROR, "Tier is empty"))
+                    .build();
+        }
+        if (subscriber == null) { //todo: implement permission
+            subscriber = username;
+        }
+
+        try {
+            APIConsumer apiConsumer = APIManagerFactory.getInstance().getAPIConsumer(username);
+            Application application = ApplicationMappingUtil.fromDTOtoApplication(body, new Subscriber(subscriber));
+            String status = apiConsumer.addApplication(application, subscriber);  //todo: use "status" properly
+            return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, status)).build();
+        } catch (APIManagementException e) {
+            throw new InternalServerErrorException(e);
+        }
     }
+
     @Override
-    public Response applicationsApplicationIdGet(String applicationId,String accept,String ifNoneMatch,String ifModifiedSince){
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
+    public Response applicationsApplicationIdGet(String applicationId, String subscriber, String accept,
+            String ifNoneMatch, String ifModifiedSince) {
+        //todo: need to be improved. Should avoid iteration through all applications.       
+        String username = CarbonContext.getThreadLocalCarbonContext().getUsername();
+        if (subscriber == null) { //todo: implement permission
+            subscriber = username;
+        }
+        String groupId = "";  //todo: add to model
+        int applicationID = Integer.parseInt(applicationId);
+        ApplicationDTO applicationDTO = null;
+        try {
+            APIConsumer apiConsumer = APIManagerFactory.getInstance().getAPIConsumer(username);
+            Application[] applications = apiConsumer.getApplications(new Subscriber(subscriber), groupId);
+            for (Application application : applications) {
+                if (application.getId() == applicationID) {
+                    applicationDTO = ApplicationMappingUtil.fromApplicationtoDTO(application);
+                }
+            }
+            if (applicationDTO != null) {
+                return Response.ok().entity(applicationDTO).build();
+            } else {
+                throw new NotFoundException();
+            }
+        } catch (APIManagementException e) {
+            throw new InternalServerErrorException(e);
```

Comment:
Already fixed error handling in #1303 (https://github.com/wso2/carbon-apimgt/commit/10b904553097e950f881465d125ef962ac32a3a3)